boards: arm: Add phyBOARD-Lyra AM62x M4F

dnltz opened the pull request:
This is a copy of the TI AM62x SK EVM board #58722 file to support the M4F on the phyBOARD-Lyra AM62x, which is PHYTEC's development kit for the phyCORE-AM62x SoM.

dnltz committed:
dts: arm: ti: Rename AM62x M4F base device-tree

Rename the base device-tree for the AM62x M4F from a SK EVM specific
to a more generic name since this DT describes the M4F subsystem in
the AM62x SoC.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

dnltz committed:
boards: arm: Add AM62x M4F phyBOARD-Lyra

This is a copy of the TI AM62x SK EVM board file to support the
M4F on the phyBOARD-Lyra AM62x, which is PHYTEC's development kit
for the phyCORE-AM62x SoM.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

vaishnavachath commented:
@dnltz Thank you for the pull request, you may need to compress the image(use webp?) to meet with the image size requirements.

vaishnavachath commented on dts/arm/ti/am62x_phycore_m4.dtsi:
```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2023 PHYTEC Messtechnik GmbH
+ * Author: Daniel Schultz <d.schultz@phytec.de>
+ *
```
Do we need this separately for this platform? 
The original PR named the corresponding file `dts/arm/ti/am62x_sk_m4.dtsi` while it should have been `am62x_m4.dtsi` , can you reuse the same file?

dnltz replied:
Actually we do not since this file should describe the M4F subsystem which is the same in all AM62x boards. Changed the name to `am62x_m4.dtsi` and update the sk board device-tree file.

KarthikL1729 reviewed (commented):
Should the board directory not be named `phyboard_lyra_am62x_m4`? That is usually the convention if I understand correctly.

vaishnavachath commented on boards/arm/ti_am62x_phyboard_m4/doc/index.rst:
```diff
@@ -0,0 +1,149 @@
+********
+
+The board can using remoteproc, and uses the OpenAMP resource table to accomplish this.
+
```
nitpick: "can be loaded" , also instead of board it would be better to mention M4 core.

vaishnavachath commented:
> Should the board directory not be named `phyboard_lyra_am62x_m4`? That is usually the convention if I understand correctly.
Yes @KarthikL1729, that is correct. @dnltz can you please update that as well?

dnltz commented:
@KarthikL1729 @vaishnavachath are you okay if I rename my board to `am62x_phyboard_lyra_m4` and drop the `ti_` appendix for both boards? By having the same `am62x_` appendix, all AM62x boards will be next to each other. I think this will increase the visibility and makes the maintenance easier. Otherwise, boards will spread over the entire `arm/` directory.

vaishnavachath commented:
> @KarthikL1729 @vaishnavachath are you okay if I rename my board to `am62x_phyboard_lyra_m4` and drop the `ti_` appendix for both boards? By having the same `am62x_` appendix, all AM62x boards will be next to each other. I think this will increase the visibility and makes the maintenance easier. Otherwise, boards will spread over the entire `arm/` directory.

@dnltz I am okay with that proposal, another suggestion is all AM62X M4 board definitions under single  directory (am62x_m4) with a common DTSI for shared description, see similar approach in `boards/arm/efr32_radio/`

dnltz committed:
board: arm: Move AM62x M4 SK into own subdirectory

Move the AM62x M4 SK board into a am62x_m4 subdirectory to collect
all AM62x M4 boards there.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

dnltz committed:
MAINTAINERS: Force AM6 boards to start with 'am6'

Force all AM6-based boards to start with 'am6' which allows to
place all boards for the same SoC next to each other. The 'ti_'
prefix would end in spreading boards all over the board directory.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

dnltz commented:
@vaishnavachath @KarthikL1729 I added your feedback and reworked the structure. They share now the same board directory and have a common AM62x M4 index page.

Regarding the MAINTAINERS changes ... my plan is to rework the current AM62x A53 board soon too and align it to the current name schema which would make the entry for the `phycore_am6*` obsolete.

vaishnavachath reviewed (dismissed):
@dnltz changes looks good to me. @cfriedt confirming that this kind of directory structure is permitted, seen something similar at `boards/arm/efr32_radio/`, these platforms share a lot of common changes, and having these in same directory will make it easier to maintain and avoid duplication and also future fixes to be propagated to all platforms.

dnltz commented on boards/arm/am62x_m4/doc/index.rst:
```diff
@@ -0,0 +1,38 @@
+for the M4F subsystem.
+
+Currently the following devices are supported:
+
```
Changed that to hardware platforms, thanks!

vaishnavachath commented:
@cfriedt Can you please help approve this PR? also does the directory structure(keeping both related boards under same folder) look okay to you?

DhruvaG2000 commented on boards/arm/am62x_m4/doc/am62x_m4_phyboard_lyra.rst:
```diff
@@ -0,0 +1,148 @@
+
+.. code-block:: console
+
+   setenv overlays k3-am62-phyboard-lyra-rpmsg.dtbo
```
interesting, what does the overlay do? Why is it required? Perhaps you can add some link to the source of this overlay in Documentation?

DhruvaG2000 replied:
I am assuming that you probably "disable" MCU core in your default DT? Any reason to do that by default? Maybe we can just enable it from default DT instead of needing an overlay?
